app/controllers/admin/watched_words_controller.rb

```diff
@@ -23,6 +24,7 @@ def create
   def destroy
     watched_word = WatchedWord.find_by(id: params[:id])
     raise Discourse::InvalidParameters.new(:id) unless watched_word
+    StaffActionLogger.new(current_user).log_watched_words_deletion(watched_word)
```

Comment:
I think we should wrap this in a DB transaction or at least only log after `watched_word.destroy!` does not raise an error.